red/runtime/util.js

```diff
@@ -328,6 +328,11 @@ function evaluateNodeProperty(value, type, node, msg) {
     return value;
 }
 
+function normaliseRegisterTypeName(name) {
```

Comment:
I suggest the function should be called`normaliseNodeTypeName`.

The current approach to remove characters is a good start, but I think it should honour camel-case: 'http-request' should become 'httpRequest', not 'httprequest' as it would with the current implementation.